picohttpparser.c

```diff
@@ -429,15 +429,20 @@ static const char *parse_response(const char *buf, const char *buf_end, int *min
     }
     PARSE_INT_3(status);
 
-    /* skip space */
-    if (*buf++ != ' ') {
-        *ret = -1;
-        return NULL;
+    int has_sp = *buf == ' ';
+    if (has_sp) {
+        /* skip space */
+        ++buf;
     }
     /* get message */
     if ((buf = get_token_to_eol(buf, buf_end, msg, msg_len, ret)) == NULL) {
         return NULL;
     }
+    if (!has_sp && *msg_len != 0) {
+        /* garbage found after status code */
+        *ret = -1;
+        return NULL;
+    }
```

Comment:
Instead of checking for whitespace then calling get_token_to_eol, how about calling get_token_to_eol then checking that either the length of the token was zero or that the first character was a whitespace (which then gets popped)?

I think doing that way might simplify the code, by gathering the code related to this whitespace handling into one place.

Reply:
thank you, done in https://github.com/h2o/picohttpparser/pull/47/commits/75820a54212de7ba5c6d7b35a0204f3b44a3ab77